{{cookiecutter.github_repository}}/{{cookiecutter.main_module}}/users/auth/serializers.py

```diff
@@ -43,3 +45,70 @@ def get_name(self, obj):
 
     def get_auth_token(self, obj):
         return get_token_for_user(obj, "authentication")
+
+
+class PasswordChangeSerializer(serializers.Serializer):
+    current_password = serializers.CharField(required=True)
+    new_password = serializers.CharField(required=True)
+
+    error_messages = {
+        'invalid_password': 'Invalid password.'
+    }
+
+    def validate_current_password(self, value):
+        if not self.context['request'].user.check_password(value):
+            raise serializers.ValidationError(self.error_messages['invalid_password'])
+        return value
+
+    def validate_new_password(self, value):
+        # https://docs.djangoproject.com/en/2.0/topics/auth/passwords/#django.contrib.auth.password_validation.validate_password
+        password_validation.validate_password(value)
+        return value
+
+
+class PasswordResetSerializer(serializers.Serializer):
+    email = serializers.EmailField(required=True)
+
+    error_messages = {
+        'email_does_not_exist': "No user with the specified email found"
+    }
+
+    def validate_email(self, value):
+        user = User.objects.filter(email__iexact=value, is_active=True).first()
+        if user is None:
+            raise serializers.ValidationError(self.error_messages['email_does_not_exist'])
+        # use this as serializer.user in the view
+        self.user = user
+        return value
+
+
+class PasswordResetConfirmSerializer(serializers.Serializer):
+    new_password = serializers.CharField(required=True)
+    user_id = serializers.CharField(required=True)
```

Comment:
Shall we call it `uid` instead of `user_id`? It's not really a user_id but a encoded user_id.